### PR TITLE
Fix random stuff. Backend + Frontend

### PR DESCRIPTION
**Pull request by odinuge**

### CODE REVIEW

**Comment by odinuge** [comment]
This will require a `Committee.objects.delete()` and `Membership.objects.delete()` before migrating... :smile:

**Comment by cherrybeans** [comment]
Failing tests :(

File: committee_admissions/admissions/authentication.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from rest_framework import authentication
+
+
+class SessionAuthentication(authentication.SessionAuthentication):
+    """
+    This class is needed, because REST Framework's default SessionAuthentication does never return
+    401's, because they cannot fill the WWW-Authenticate header with a valid value in the 401
+    response. As a result, we cannot distinguish calls that are not unauthorized (401 unauthorized)
+    and calls for which the user does not have permission (403 forbidden).
+
```
Comment (cherrybeans) [inline committee_admissions/admissions/authentication.py]: Explains some things :+1:

File: committee_admissions/admissions/models.py [hunk 2]
```diff
@@ -118,11 +94,11 @@ class CommitteeApplication(TimeStampModel):
 
 class Membership(models.Model):
     user = models.ForeignKey(User, on_delete=models.CASCADE)
-    abakus_group = models.ForeignKey(Group, on_delete=models.CASCADE)
+    committee = models.ForeignKey(Committee, on_delete=models.CASCADE)
```
Comment (cherrybeans) [inline committee_admissions/admissions/models.py]: Good job fixing all the concerns we had :+1:

File: committee_admissions/oauth.py [hunk 3]
```diff
@@ -48,40 +49,76 @@ def user_data(self, access_token, *args, **kwargs):
...
     """ This will run after the social auth pipelies succeeds """
     if not user:
         return
-    user_memberships = kwargs['response']['memberships']
-    abakus_groups = kwargs['response']['abakusGroups']
 
-    # Remove old memberships before creating the new ones
-    Membership.objects.filter(user=user).delete()
+    group_data = parse_group_data(kwargs['response'])
+    with transaction.atomic():
```
[the remark this answers is not in the document]
Comment (odinuge) [reply committee_admissions/oauth.py]: If one of the django operations inside the `with transaction`, all the previously operations are rolled back. :smile:

File: committee_admissions/admissions/views.py [hunk 4]
```diff
@@ -64,25 +70,19 @@ def get_permissions(self):
...
-        ).exists():
+        user.__class__ = LegoUser
+        if user.is_superuser:
             return super().get_queryset().prefetch_related(
                 'committee_applications', 'committee_applications__committee'
             )
 
-        membership = Membership.objects.filter(user=user, role=constants.LEADER).first()
-        if not membership:
+        if not user.is_board_member:
```
[the remark this answers is not in the document]
Comment (odinuge) [reply committee_admissions/admissions/views.py]: Now:

Leader of abakus has `user.is_superuser = True`, and board members has `user.is_board_member = True` (including leader). Each leader has `user.leader_of_committee`, that returns the committee

File: committee_admissions/admissions/tests/test_api.py [hunk 5]
```diff
@@ -68,17 +65,15 @@ def test_committee_leader_can_edit_committee(self):
...
+        abakus_leader = LegoUser.objects.create(username="bigsupremeleader", is_superuser=True)
 
         self.client.force_authenticate(user=abakus_leader)
 
         res = self.client.patch(
             reverse('committee-detail', kwargs={'pk': self.arrkom.pk}), self.edit_committee_data
         )
 
-        self.assertEqual(res.status_code, status.HTTP_403_FORBIDDEN)
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
```
[the remark this answers is not in the document]
Comment (odinuge) [reply committee_admissions/admissions/tests/test_api.py]: Yups

File: committee_admissions/admissions/tests/test_api.py [hunk 6]
```diff
@@ -172,38 +180,58 @@ def setUp(self):
...
+        res = self.client.post(
+            reverse('userapplication-list'), annas_application_data, format='json'
+        )
+
+        self.assertEqual(res.status_code, status.HTTP_201_CREATED)
+
+        pk_res = res.json()['pk']
+
+        # Application registered as bob
+        self.assertEqual(UserApplication.objects.get(pk=pk_res).user, self.pleb_bob)
```
[the remark this answers is not in the document]
Comment (odinuge) [reply committee_admissions/admissions/tests/test_api.py]: Yups.

The commenting is kinda bad, but you are right.

File: committee_admissions/admissions/serializers.py [hunk 7]
```diff
@@ -115,15 +112,14 @@ class Meta:
...
+        fields = ('text', 'pk')
 
     def create(self, validated_data):
         user = validated_data.pop('user')
-        print(user, vars(user))
 
         text = validated_data.pop('text')
 
-        admission = [obj for obj in Admission.objects.all() if obj.is_open][0]
+        admission = Admission.objects.filter(application_deadline__gt=timezone.now()).first()
```
[the remark this answers is not in the document]
Comment (odinuge) [reply committee_admissions/admissions/serializers.py]: Yeah. Guess I can revert this. :upside_down_face:

File: committee_admissions/templates/index.html [hunk 8]
```diff
@@ -23,9 +23,11 @@
     </noscript>
     <div id="root"></div>
     <script>
-    window.django = {
+    window.__DJANGO__ = {
```
Comment (odinuge) [inline committee_admissions/templates/index.html]: This object can be reached by using:
`import djangoData from "src/utils/djangoData";`

Then `djangoData` is `{user: ...}`

File: committee_admissions/settings/local.py [hunk 9]
```diff
@@ -1,13 +0,0 @@
-from .development import *
```
Comment (cherrybeans) [inline committee_admissions/settings/local.py]: Deleting this file means import error for everyone that does not have it, so if the point is to delete it entirely, the `__init__.py` file in settings must be updated from using `local` to using `development`.

**Comment by cherrybeans** [comment]
Is the massively long migrate logging really necessary? 

screenshot from 2018-08-23 08-46-02

**Comment by cherrybeans** [comment]
Now everyone must add their own `local.py` file that contains the line `from development import *`, then flush their database, migrate, and make an admission in shell to avoid frontend crashing.

**Pull request merged**